src/connection.ts

```diff
@@ -88,7 +91,12 @@ export class Connection extends EventEmitter<ConnectionEvents> {
 	private _connected = false
 	private _version = Version.v23x
 
-	constructor(private host: string, private port = 5250, autoConnect: boolean) {
+	constructor(
+		private host: string,
+		private port = 5250,
+		autoConnect: boolean,
+		private _getRequestForResponse: (response: Response<any>) => SentRequest | undefined
```

Comment:
I think this lookup is unnecessary, it'd be more efficient to just save the expected deserializer on the `Connection` class in the `sendCommand` method and I think that also feels a bit cleaner than having a callback

Reply:
I tried to do something like you suggested, but I didn't end up with anything I liked because I ended up with 2 lookups of the previously sent `request` anyway, which was kinda awkward too.

Reply:
I can see your point. I've experimented a bit here: https://github.com/SuperFlyTV/casparcg-connection/commit/38a9e87b39a3a5e322518e7995fc7325b674dc20

It's a trade off between adding more dependency between the classes vs doing some extra storage. Hard to say which is lower complexity.